saleor/graphql/order/tests/test_order.py

```diff
@@ -4749,12 +4749,370 @@ def test_order_refund(staff_api_client, permission_manage_orders, payment_txn_ca
+        "id": order_id,
+        "paymentsToRefund": [
+            {
+                "paymentId": graphene.Node.to_global_id(
+                    "Payment", payment_txn_captured.id
+                ),
+                "amount": amount,
+            }
+        ],
+    }
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_orders]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["orderRefund"]["order"]
+    order.refresh_from_db()
+    assert data["status"] == order.status.upper()
+    assert data["paymentStatus"] == OrderPaymentStatusEnum.FULLY_REFUNDED.name
+    payment_status_display = dict(OrderPaymentStatus.CHOICES).get(
+        OrderPaymentStatus.FULLY_REFUNDED
+    )
+    assert data["paymentStatusDisplay"] == payment_status_display
+    assert data["isPaid"] is False
+
+    refund_order_event = order.events.filter(
+        type=order_events.OrderEvents.PAYMENT_REFUNDED
+    ).first()
+    assert refund_order_event.parameters["amount"] == str(amount)
+
+    refunded_fulfillment = order.fulfillments.filter(
+        status=FulfillmentStatus.REFUNDED
+    ).first()
+    assert refunded_fulfillment
+    assert refunded_fulfillment.total_refund_amount == payment_txn_captured.total
+    assert refunded_fulfillment.shipping_refund_amount is None
+
+
+def test_order_refund_fails_if_both_amount_and_payments_to_refund_specified(
+    staff_api_client, permission_manage_orders, payment_txn_captured
+):
+    # given
+    order = payment_txn_captured.order
+    query = """
+        mutation refundOrder(
+            $id: ID!,
+            $amount: PositiveDecimal!,
+            $paymentsToRefund: [OrderPaymentToRefundInput]
+        ) {
+            orderRefund(id: $id, amount: $amount, paymentsToRefund: $paymentsToRefund) {
+                order {
+                    paymentStatus
+                    paymentStatusDisplay
+                    isPaid
+                    status
+                },
+                errors {
+                    field,
+                    message,
+                    variants,
+                    code
+                }
+            }
+        }
+    """
+    order_id = graphene.Node.to_global_id("Order", order.id)
+    amount = float(payment_txn_captured.total)
+    variables = {
+        "id": order_id,
+        "amount": amount,
+        "paymentsToRefund": [{"paymentId": "asd", "amount": amount}],
+    }
+
+    # when
+    message = "Either amount or payments_to_refund should be specified."
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_orders]
+    )
+    content = get_graphql_content_from_response(response)
+    order_refund_data = content["data"]["orderRefund"]
+
+    # then
+    assert len(order_refund_data["errors"]) == 2
+    assert order_refund_data["errors"][0]["message"] == message
+    assert order_refund_data["errors"][1]["message"] == message
```

Comment:
In tests we do not compere messages, as they are often changing. We compare `code` and `field` error fields.

Reply:
I agree regarding checking `code` and `field`  in tests, will add it.
Regarding the messages, they are actually checked in other tests too (in quite many of them) + if they are displayed on the frontend, then I think there is no harm making sure that the sentences don't change accidentally?

What do you think? Otherwise, I'm happy to remove the check if it's the policy.